testnet/advancedCounter.js

```diff
@@ -59,34 +56,41 @@ function sleep(ms) {
 
       // keep the contract funding constant
 
-      const newLockingScript = advCounter.getNewStateScript({counter: i +1})
+      const newLockingScript = advCounter.getNewStateScript({ counter: i + 1 })
+
+      const unlockingTx = new bsv.Transaction();
+
+      unlockingTx.addInput(createInputFromPrevTx(prevTx))
+        .from(await fetchUtxos(privateKey.toAddress()))
+        .addOutput(new bsv.Transaction.Output({
+          script: newLockingScript,
+          satoshis: amount,
+        }))
+        .change(privateKey.toAddress())
+        .setInputScript(0, (tx, _) => {
+          const preimage = getPreimage(
+            tx,
+            advCounter.lockingScript,
+            amount,
+            0,
+            Signature.SIGHASH_ANYONECANPAY |
+            Signature.SIGHASH_SINGLE |
+            Signature.SIGHASH_FORKID
+          );
+          return advCounter
+            .increment(new SigHashPreimage(toHex(preimage)))
+            .toScript();
+        })
+        .sign(privateKey);
```

Comment:
How does this sign every input, but input 0?

Reply:
call sign for every input,  but return empty for contract input.

![image](https://user-images.githubusercontent.com/995909/139772344-b9b783b2-b3a1-4e58-a4f3-adb1983e657a.png)